cmd/signature-v4-utils.go

```diff
@@ -157,3 +157,15 @@ func extractSignedHeaders(signedHeaders []string, reqHeaders http.Header) (http.
 
 	return extractedSignedHeaders, ErrNone
 }
+
+// Sequential spaces regular expression
+var seqSpaces = regexp.MustCompile("  +")
+
+// Trim leading and trailing spaces and replace sequential spaces with one space, following Trimall()
+// in http://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+func sigV4TrimAll(input string) string {
+	input = strings.TrimPrefix(input, " ")
+	input = strings.TrimSuffix(input, " ")
+	replaced := seqSpaces.ReplaceAllString(input, " ")
+	return replaced
+}
```

Comment:
The documentation ("spec") seems to be a bit vague `The Trimall function removes excess white space before and after values, and converts sequential spaces to a single space.` - so what do we do in case prefix is whitespace other than spaces (like tabs) - e.g. in case input is `  \tabc\t sc\t\t` - what should be used in the canonical request is a bit unclear. Maybe we can refer to what an AWS released library does in these cases.